cli/cmd/plugin/unmanaged-cluster/cluster/minikube.go

```diff
@@ -27,15 +27,54 @@ type MinikubeClusterManager struct {
 }
 
 type MinikubeProfile struct {
-	Name   string `json:"Name"`
-	Status string `json:"Status"`
+	Name   string         `json:"Name"`
+	Status string         `json:"Status"`
+	Config MinikubeConfig `json:"Config"`
 }
 
 type MinikubeProfiles struct {
 	Invalid []MinikubeProfile `json:"invalid"`
 	Valid   []MinikubeProfile `json:"valid"`
 }
 
+type MinikubeConfig struct {
+	Driver string `json:"Driver"`
+}
+
+// Get retrieves cluster information. An error is returned if no cluster is
+// found or if there is a failure communicating with minikube.
+func (mkcm *MinikubeClusterManager) Get(clusterName string) (*KubernetesCluster, error) {
+	var resolvedStatus string
+
+	mkp, err := mkcm.getProfile(clusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find a cluster profile with name %s via minikube", clusterName)
+	}
+
+	// "Running" and "Stopped" are known-good status from minikube. Any other value should be
+	// considered unknown.
+	switch mkp.Status {
+	case "Running":
+		resolvedStatus = StatusRunning
+	case "Stopped":
+		resolvedStatus = StatusStopped
+	default:
+		resolvedStatus = StatusUnknown
+	}
+
+	kc := &KubernetesCluster{
+		Name: clusterName,
+		// TODO(joshrosso): We should consider this field in future
+		// work. Perhaps when we expose get at the CLI level, we could
+		// do a command like `tanzu uc get ${CLUSTER_NAME} --kubeconfig` and return this value.
```

Comment:
This would be great.

And we've already set ourselves up for success since we persist the kubeconfig in the unmanaged-cluster config directory for that named cluster.